File: csharp/2021/day1/day1.cs
using System;
using System.IO;

namespace Advent2021
{
    class Day1
    {
        static void Main(string[] args)
        {
            string[] lines = File.ReadAllLines("./day1.in");
            int[] data = new int[lines.Length];
            for (int i = 0; i < lines.Length; i++)
            {
                data[i] = Int32.Parse(lines[i]);
            }

            int p1Count = 0, p2Count = 0;

            for (int i = 0; i < data.Length - 1; i++) {       
                if (data[i] < data[i+1]) {
                    p1Count++;
                }
            }

            for (int i = 0; i < data.Length - 3; i++) {       
                if (data[i] + data[i+1] + data[i+2] < data[i+1] + data[i+2] + data[i+3]) {
                    p2Count++;
                }
            }

            Console.WriteLine("Part 1: {0}\nPart 2: {1}", p1Count, p2Count);
        }
    }
}

File: rust/2021/day1/day1.rs
use std::fs;

fn main() {
    let raw = fs::read_to_string("./day1.in")
        .expect("Could not read file!");
    let lines: Vec<u16> = raw.lines().map(|x| x.parse::<u16>().unwrap()).collect();

    let mut part_1 = 0;
    let mut part_2 = 0;

    for i in 1..lines.len() {
        part_1 += (lines[i-1] < lines[i]) as u16;
    }

    for i in 3..lines.len() {
        part_2 += (lines[i-3] + lines[i-2] + lines[i-1] < lines[i-2] + lines[i-1] + lines[i]) as u16;
    }

    println!("Part 1: {}\nPart 2: {}", part_1, part_2);
}